src/main/java/chopchop/model/UsageList.java

```diff
@@ -89,30 +89,30 @@ public int getUsageCount() {
         requireNonNull(lowerBound);
         return this.usages.stream()
             .filter(x-> x.isAfter(lowerBound))
-            .sorted(comparator)
             .collect(Collectors.toList());
     }
 
     public List<T> getUsagesBefore(LocalDateTime upperBound) {
         requireNonNull(upperBound);
         return this.usages.stream()
             .filter(x-> x.isBefore(upperBound))
-            .sorted(comparator)
```

Comment:
why you not sorting here?

Reply:
I change it to be sorted  `getUsagesBetween`  instead. Because `getUsagesBefore` is not called by Model. Should I still sort in `getUsagesBefore`?

Reply:
if these two methods are not called by the model, then you should make them private? so nobody outside can get a list of unsorted usages

Reply:
True. I decided to move it back down so that `getUsagesBetween` has less things to do